ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/properties/FontSpec.kt

```diff
@@ -58,9 +60,50 @@ internal sealed interface FontSpec {
 internal fun Map<FontAlias, FontsConfig>.determineFontSpecs(
     resourceProvider: ResourceProvider,
 ): Map<FontAlias, FontSpec> {
+    val resourceFontFamilies = values
+        .toSet()
+        .map { it.android }
+        .filterIsInstance<FontInfo.Name>()
+        .mapNotNull { fontInfo -> fontInfo.family?.let { family -> fontInfo to family } }
+        .groupBy({ it.second }, { it.first })
+        .mapValues { (_, fontInfos) ->
+            val resourceIdsSeen = mutableSetOf<Int>()
+            fontInfos.mapNotNull { fontInfo ->
+                resourceProvider.getResourceIdentifier(name = fontInfo.value, type = "font")
+                    .takeIf { it != 0 && it !in resourceIdsSeen }
+                    ?.also { resourceIdsSeen.add(it) }
```

Comment:
No need to continue here with multiple fonts to check if they are pointing to the same resource id.